engine/class_modules/sc_death_knight.cpp

```diff
@@ -6295,6 +6311,11 @@ struct unholy_blight_dot_t : public death_knight_spell_t
     tick_may_crit = background = true;
     may_miss = may_crit = hasted_ticks = false;
   }
+
+  void impact(action_state_t* state) override
```

Comment:
Unholy Blight should probably be reworked into a buff that applies the stacking UB dot (tied to the debuff) and the VP dot every tick, similarly to Remorseless Winter.